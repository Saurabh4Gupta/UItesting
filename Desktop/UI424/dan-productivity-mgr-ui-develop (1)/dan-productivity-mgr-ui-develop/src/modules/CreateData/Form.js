/* eslint-disable react/prop-types */
import React from 'react';

import {
  Select,
  FormField,
  TextInput,
  Textarea,
  DateInput,
  Dropzone,
  Stack,
  Box,
  ChipInput,
} from '@dentsu-ui/components';
import constant from '../../utils/constant';
import Loader from '../../components/loading';

const Form = (props) => {
  const { MAX_FILE_SIZE, ALLOWED_FILE_TYPES } = constant;
  const {
    values,
    handleChange,
    handleSelectField,
    errors,
    cmsData,
    options,
    monthOptions,
    reportingYear,
    forecastOptions,
    userList,
    handleOwners,
    setFiles,
    loading,
    isEdit,
    dueDate,
  } = props;

  if (loading) return <Loader />;
  const now = new Date();
  const maxDate = new Date();
  maxDate.setFullYear(now.getFullYear() + 1);
  return (
    <>
      <FormField
        label={cmsData.localMarketLabel}
        {...(errors.localMarket && !isEdit
          ? { error: errors.localMarket }
          : {})}
        hint={cmsData.localMarketHint}
        isDisabled={isEdit}
      >
        <Select
          placeholder={cmsData.selectPlaceHolder}
          name="localMarket"
          options={options}
          onChange={(selected, event) => handleSelectField(selected, event)}
          value={options.find((key) => key.value === values.localMarket.value)}
        />
      </FormField>
      <FormField
        {...(errors.name ? { error: errors.name } : {})}
        label={cmsData.nameLabel}
        hint={cmsData.nameHint}
      >
        <TextInput
          name="name"
          maxLength={255}
          value={values.name}
          onChange={handleChange}
        />
      </FormField>
      <FormField
        label={cmsData.briefLabel}
        {...(errors.briefing ? { error: errors.briefing } : {})}
        hint={cmsData.briefHint}
      >
        <Textarea
          maxLength={4000}
          isFullWidth="true"
          placeholder=""
          name="briefing"
          value={values.briefing}
          onChange={handleChange}
        />
      </FormField>
      <Stack flex="row">
        <Box width="350px">
          <FormField
            label={cmsData.reportingYearLabel}
            {...(errors.reportingYear ? { error: errors.reportingYear } : {})}
            hint={cmsData.reportingYearHint}
          >
            <Select
              name="reportingYear"
              options={reportingYear}
              value={reportingYear.find(
                (key) => key.value === values.reportingYear,
              )}
              placeholder={cmsData.selectPlaceHolder}
              onChange={(selected, event) => {
                handleSelectField(selected, event);
              }}
            />
          </FormField>
        </Box>
      </Stack>
      <Stack flex="row">
        <Box mr="30px" width="50%">
          <FormField
            label={cmsData.actualDataLabel}
            {...(errors.actualData ? { error: errors.actualData } : {})}
            hint={cmsData.actualDataHint}
          >
            <Select
              name="actualData"
              options={monthOptions}
              value={monthOptions.find(
                (key) => key.value === values.actualData,
              )}
              placeholder={cmsData.selectPlaceHolder}
              onChange={(selected, event) => {
                handleSelectField(selected, event);
              }}
            />
          </FormField>
        </Box>
        <Box width="50%">
          <FormField
            isDisabled={!forecastOptions}
            label={cmsData.forecastDataLabel}
            {...(errors.forecastData ? { error: errors.forecastData } : {})}
            hint={cmsData.forecastDataHint}
          >
            <Select
              name="forecastData"
              placeholder={cmsData.selectPlaceHolder}
              value={monthOptions.find(
                (key) => key.value === values.forecastData,
              )}
              options={forecastOptions}
              onChange={(selected, event) => handleSelectField(selected, event)}
            />
          </FormField>
        </Box>
      </Stack>
      <FormField
        label={cmsData.templateFileLabel}
        hint={cmsData.templateFileHint}
        {...(errors.file ? { error: errors.file } : {})}
      >
        <Dropzone
          allowMultiple={false}
          allowReorder
          onUpdateFiles={(fileItems) => {
            setFiles(fileItems);
          }}
          maxFiles={1}
          maxFileSize={MAX_FILE_SIZE}
          // onAddFileStart={(e) => handleUploadFile(e)}
          acceptedFileTypes={ALLOWED_FILE_TYPES}
          labelMaxFileSizeExceeded={cmsData.uploadFileLargeMessage}
          labelFileTypeNotAllowed={cmsData.labelFileTypeNotAllowed}
          // files={values.file}
        />
      </FormField>
      <FormField
        {...(errors.dueDate ? { error: errors.dueDate } : {})}
        label={cmsData.dueDateLabel}
        hint={cmsData.dueDateHint}
      >
        <DateInput
          placeholder={cmsData.datePlaceHolder}
          dateFormat="YYYY/MM/DD"
          parseDate={(date) => new Date(date).toUTCString()}
          onChange={(date) => handleSelectField(date, { name: 'dueDate' })}
          minDate={new Date()}
          maxDate={new Date(maxDate)}
          value={isEdit ? new Date(values.dueDate || '') : dueDate}
        />
      </FormField>

      <FormField
        label={cmsData.assignToLabel}
        {...(errors.assignTo ? { error: errors.assignTo } : {})}
        hint={cmsData.assignToHint}
      >
        <ChipInput
          options={userList}
          isMulti
          name="assignTo"
          value={userList.filter((obj) => values.assignTo.includes(obj.value))}
          onChange={(selected, event) => handleOwners(selected, event)}
        />
      </FormField>
    </>
  );
};

export default Form;
