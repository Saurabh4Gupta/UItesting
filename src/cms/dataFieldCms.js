export const dataFieldCms = {
  createNewDataRequest: 'Create new data request',
  editDataRequest: 'Edit data request',
  localMarketLabel: 'Local market',
  localMarketHint: 'Choose the market you wish to create this data request for',
  nameLabel: 'Data request name',
  backLabel:'Back to clients',
  nameHint:
    "Make sure it's clear to help others identify it.Max 255 characters",
  briefLabel: 'Briefing',
  briefHint:
    'Tell us in detail what this data request is for, max 4000 characters',
  templateFileLabel: 'Add tracker template file',
  templateFileHint:
    'Only .xls, .xlsx, .csv, .xlsb, .xlsm, .xlt, .xltm, .xltx files in 20mb or less',
  uploadFileLabel:
    'To upload a new file,drag and drop you file below or browse using the link',
  uploadFileLargeMessage: 'File must be less than 20MB',
  labelFileTypeNotAllowed: 'File must be of type .xls, .xlsx, .csv, .xlsb, .xlsm, .xlt, .xltm, .xltx',
  uploadButtonText:'Upload new file',
  actualDataLabel: 'Actual data',
  versionHistoryTitle:'Version history',
  actualDataHint: 'Number of months of actual productivity data',
  forecastDataLabel: 'Forecast data',
  forecastDataHint: 'Number of months of forecasted productivity data',
  dueDateLabel: 'Due date',
  dueDateHint: 'Enter the date that you need this data request by',
  assignToLabel: 'Assign to',
  assignToHint: `Only the person or team you assign this to will have access. Once you click 'Create' ,
  they will be sent email notifications asking them to fill in the data request.`,
  create: 'Create',
  upload: 'Upload',
  cancel: 'Cancel',
  productivityDatarequestHeading: 'Productivity data requests',
  productivityDatarequestCaption: ' The ongoing and complete productivity data requests created for this client',
  selectPlaceHolder: 'Please select',
  datePlaceHolder: 'Date',
  viewDetails: 'View Details',
  viewClientsHeading: 'My clients',
  viewClientsCaption: 'Choose the client you wish to view',
  completeLabel: 'Complete',
  ongoingLabel: 'Ongoing',
  emptyProductivityDatarequestCaption:
    'There are currently no active Productivity data requests',
  emptyLabel: 'Create new data request',
  edit: 'Edit',
  actual: 'Actuals',
  forecast: 'Forecast',
  overdue: 'Overdue',
  ongoing: 'Ongoing',
  complete: 'Complete',
  lastUpdate: 'Last Update',
  moveToComplete: 'Move to Complete',
  moveToOnGoing: 'Move to Ongoing',
  delete: 'Delete',
  deletePopConfimationMessage:'Once deleted, this data request will no longer be available.',
  deleteButtonCancelMessage:'No, Cancel',
  deleteButtonDeleteMessage:'Yes, Delete',
  popUpTitle:'Are you sure you want to delete?',
  emptyCompletedProductivityDatarequestCaption:
    'No Productivity data requests have been moved to this section yet.',
  editRequest: 'Edit request',
  save: 'Save',
  productivityRequest: 'Productivity request',
  showMore: 'Show more',
  reportingYear: 'Reporting year',
  productivityDataRequested: 'Productivity data requested',
  actualCapitalized: 'Actual',
  forecastCapitalized: 'Forecast',
  dueBy: 'Due by',
  assignedTo: 'Assigned to',
  trackerTemplate: 'Tracker template',
  close: 'Close',
  reportingYearLabel: 'Reporting year',
  reportingYearHint: 'The reporting year this data request is for',
  toastRequestCreated: 'Data request created',
  toastRequestEdited: 'Data request edited',
  toastRequestDeleted: 'Data request deleted',
  toastFileUploaded: 'File uploaded',
  toastMovedToOngoing: 'Data request moved to ongoing',
  toastMovedToComplete: 'Data request moved to complete',
  currentVersionText:'current version',
  loadMoreText:'Load more',
  uploadFileErrorMessage:'Upload tracker template',
  moveToCompleteConfirmation: 'Are you sure you want to move this to complete?',
  moveToCompleteDescription: 'You will not be abale to make any more edits to this request unless it is moved back to "Ongoing".',
  deletePopUpConfirmation: 'Are you sure you want to delete?',
  deletePopUpDescription: 'Once deleted, this data request will no longer be available.',
  noCancel: 'No, cancel',
  yesContinue: 'Yes, continue',
  yesDelete: 'Yes, delete',
}
