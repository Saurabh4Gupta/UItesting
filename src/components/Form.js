/* eslint-disable react/prop-types */
import React from 'react';

import {
  FormField,
  Dropzone
} from '@dentsu-ui/components';
import constant from '../utils/constant';

const Form = (props) => {
  const { MAX_FILE_SIZE, ALLOWED_FILE_TYPES } = constant;
  const { files, handleFileChange, errors, cmsData } = props;
 // const [files, setFiles] = useState([]);
 console.log(files,"++++++++++",handleFileChange,"+",errors)
  const handleInit = () => {
    console.log('Dropzone instance has initialised', files);
  };

  return <>
        <FormField
            label={cmsData.uploadFileLabel}
            hint={cmsData.templateFileHint}
            {...(errors ? { error: errors } : {})}
          >
            <Dropzone
              allowMultiple={false}
              onInit={() => handleInit()}
              onUpdateFiles={(fileItems) => {
                handleFileChange(fileItems)
              }}
              name="fileUpload"
              //value={values.fileUpload}
              maxFiles={1}
              maxFileSize={MAX_FILE_SIZE}
              server="./"
              acceptedFileTypes={ALLOWED_FILE_TYPES}
              dropValidation
            />
          </FormField>
    </>
  
};

export default Form;
