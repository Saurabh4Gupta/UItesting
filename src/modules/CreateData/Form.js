/* eslint-disable react/prop-types */
import React, { useState } from 'react';

import {
  Select,
  FormField,
  TextInput,
  Textarea,
  DateInput,
  Dropzone,
  Stack,
  Box,
} from '@dentsu-ui/components';
import constant from '../../utils/constant';

const Form = (props) => {
  const { MAX_FILE_SIZE, ALLOWED_FILE_TYPES } = constant;
  const { values, handleChange, handleSelectField, errors, cmsData, options, monthOptions, dueDate } = props;
  const [files, setFiles] = useState([]);
  const handleInit = () => {
    console.log('Dropzone instance has initialised', files);
  };
  return (
    <>
      <FormField
        {...(errors.localMarket ? { error: errors.localMarket } : {})}
        label={cmsData.localMarketLabel}
        hint={cmsData.localMarketHint}
      >
        <Select
          placeholder={cmsData.selectPlaceHolder}
          name="localMarket"
          options={options}
          onChange={(selected, event) => handleSelectField(selected, event)}
          value={
            options.find(key => key.value === values.localMarket.value)
          }
        />
      </FormField>
      <FormField
        {...(errors.name ? { error: errors.name } : {})}
        label={cmsData.nameLabel}
        hint={cmsData.nameHint}
      >
        <TextInput
          name="name"
          maxLength={255}
          value={values.name}
          onChange={handleChange}
        />
      </FormField>
      <FormField
        label={cmsData.briefLabel}
        {...(errors.briefing ? { error: errors.briefing } : {})}
        hint={cmsData.briefHint}
      >
        <Textarea
          maxLength={4000}
          isFullWidth="true"
          placeholder=""
          name="briefing"
          value={values.briefing}
          onChange={handleChange}
        />
      </FormField>

      <FormField
        label={cmsData.templateFileLabel}
        hint={cmsData.templateFileHint}
        {...(errors.actualData ? { error: errors.actualData } : {})}
      >
        <Dropzone
          allowMultiple={false}
          allowReorder
          onInit={() => handleInit()}
          onUpdateFiles={(fileItems) => {
            setFiles(fileItems);
          }}
          maxFiles={1}
          maxFileSize={MAX_FILE_SIZE}
          server="./"
          acceptedFileTypes={ALLOWED_FILE_TYPES}
          labelMaxFileSizeExceeded={cmsData.uploadFileLargeMessage}
          labelFileTypeNotAllowed={cmsData.labelFileTypeNotAllowed}
        />

      </FormField>
      <Stack flex="row">
        <Box mr="30px">
          <FormField
            label={cmsData.actualDataLabel}
            {...(errors.actualData ? { error: errors.actualData } : {})}
            hint={cmsData.actualDataHint}
          >
            <Select
              name="actualData"
              options={monthOptions}
              value={values.actualData}
              placeholder={cmsData.selectPlaceHolder}
              onChange={(selected, event) => {
                handleSelectField(selected, event)
              }}
            />
          </FormField>
        </Box>
        <Box>
          <FormField
            label={cmsData.forecastDataLabel}
            {...(errors.forecastData ? { error: errors.forecastData } : {})}
            hint={cmsData.actualDataHint}
          >
            <Select
              name="forecastData"
              placeholder={cmsData.selectPlaceHolder}
              value={values.forecastData}
              options={monthOptions}
              onChange={(selected, event) => {
                handleSelectField(selected, event)
              }}
            />

          </FormField>
        </Box>
      </Stack>
      <FormField
        {...(errors.dueDate ? { error: errors.dueDate } : {})}
        label={cmsData.dueDateLabel}
        hint={cmsData.dueDateHint}
      >
        <DateInput
          placeholder={cmsData.datePlaceHolder}
          dateFormat="DD/MM/YYYY"
          parseDate={(date) => new Date(date)}
          formatDate={(date) => date.toLocaleDateString()}
          onChange={(date) => handleSelectField(date, { name: 'dueDate' })}
          minDate={new Date()}
          value={dueDate}
        />
      </FormField>

      <FormField
        label={cmsData.assignToLabel}
        {...(errors.assignTo ? { error: errors.assignTo } : {})}
        hint={cmsData.assignToHint}
      >
        <Select
          menuPosition="fixed"
          options={options}
          name="assignTo"
          placeholder={cmsData.selectPlaceHolder}
          onChange={(selected, event) => handleSelectField(selected, event)}
          value={
            options.find(key => key.value === values.assignTo.value)
          }
        />
      </FormField>
    </>
  )
};

export default Form;
