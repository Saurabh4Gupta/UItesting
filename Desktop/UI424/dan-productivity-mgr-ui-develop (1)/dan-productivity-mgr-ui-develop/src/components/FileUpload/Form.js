/* eslint-disable react/prop-types */
import React from 'react';

import { FormField, Dropzone } from '@dentsu-ui/components';
import constant from '../../utils/constant';

const Form = (props) => {
  const { MAX_FILE_SIZE, ALLOWED_FILE_TYPES } = constant;
  const {
    onProcessFile,
    handleFileChange,
    errors,
    cmsData,
    setError,
    handleUploadFile,
  } = props;

  return (
    <>
      <FormField
        label={cmsData.uploadFileLabel}
        hint={cmsData.templateFileHint}
        {...errors}
      >
        <Dropzone
          allowMultiple={false}
          allowFileTypeValidation
          onAddFileStart={(e) => handleUploadFile(e)}
          onUpdateFiles={(fileItems) => {
            handleFileChange(fileItems);
          }}
          onError={(error) => {
            setError(error.main);
          }}
          onProcessFile={(error) => onProcessFile(error)}
          maxFiles={1}
          maxFileSize={MAX_FILE_SIZE}
          labelMaxFileSizeExceeded={cmsData.uploadFileLargeMessage}
          labelFileTypeNotAllowed={cmsData.labelFileTypeNotAllowed}
          acceptedFileTypes={ALLOWED_FILE_TYPES}
        />
      </FormField>
    </>
  );
};

export default Form;
