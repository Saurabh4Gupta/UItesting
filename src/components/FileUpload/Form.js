/* eslint-disable react/prop-types */
import React from 'react';

import { FormField, Dropzone } from '@dentsu-ui/components';
import constant from '../../utils/constant';

const Form = (props) => {
  const { MAX_FILE_SIZE, ALLOWED_FILE_TYPES } = constant;
  const { handleFileChange, errors, cmsData } = props;

  return (
    <>
      <FormField
        label={cmsData.uploadFileLabel}
        hint={cmsData.templateFileHint}
        {...errors}
      >
        <Dropzone
          allowMultiple={false}
          allowReorder
          onUpdateFiles={(fileItems) => {
            handleFileChange(fileItems);
          }}
          onError={(error) => {
            console.log('[FORM]', error);
          }}
          maxFiles={1}
          maxFileSize={MAX_FILE_SIZE}
          labelMaxFileSizeExceeded={cmsData.uploadFileLargeMessage}
          labelFileTypeNotAllowed={cmsData.labelFileTypeNotAllowed}
          server="./"
          acceptedFileTypes={ALLOWED_FILE_TYPES}
        />
      </FormField>
    </>
  );
};

export default Form;
