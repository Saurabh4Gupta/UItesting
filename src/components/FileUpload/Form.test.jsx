import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';
import { dataFieldCms as PageContent } from '../../cms';

describe('Success test cases for: Loader component', () => {
  let wrapper;
  describe('Snapshot test for  Loader component', () => {
    beforeEach(() => {
      const fn = jest.fn();
      const files = [];
      const error = {};
      wrapper = shallow(<Form
        handleFileChange={fn}
        files={files}
        errors={error}
        cmsData={PageContent}
      />);
    });

    it('Test to match ', () => {
     wrapper.find('Dropzone').at(0).prop('onUpdateFiles')()
    });

    it('Test to match the snapsot', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('Renders correctly', () => {
      expect(wrapper.exists()).toBe(true);
    });
  });
});

describe('Failure test cases for: Duplicate data page', () => {});
