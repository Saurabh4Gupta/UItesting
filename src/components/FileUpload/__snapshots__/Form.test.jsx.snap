// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success test cases for: Loader component Snapshot test for  Loader component Test to match the snapsot 1`] = `
<Fragment>
  <FormField
    hint="Only .xls, .xlsx, .csv, .xlsb, .xlsm, .xlt, .xltm, .xltx files in 20mb or less"
    label="To upload a new file,drag and drop you file below or browse using the link"
  >
    <Dropzone
      acceptedFileTypes={
        Array [
          "application/vnd.ms-excel",
          "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
          "application/vnd.ms-excel.sheet.binary.macroEnabled.12",
          "application/vnd.ms-excel.sheet.macroEnabled.12",
          "application/vnd.ms-excel.template.macroEnabled.12",
          "application/vnd.openxmlformats-officedocument.spreadsheetml.template",
          "text/csv",
        ]
      }
      allowFileTypeValidation={true}
      allowMultiple={false}
      dropValidation={true}
      labelFileTypeNotAllowed="File must be of type .xls, .xlsx, .csv, .xlsb, .xlsm, .xlt, .xltm, .xltx"
      labelMaxFileSizeExceeded="File must be less than 20MB"
      maxFileSize="20MB"
      maxFiles={1}
      onError={[Function]}
      onUpdateFiles={[Function]}
      server="./"
    />
  </FormField>
</Fragment>
`;
